### ISSUE TITLE
Eager loading - uses all IDs from the result

### ISSUE THREAD

**Issue by isudakoff** [yiisoft/yii2#14688]
I have a problem with eager loading.
- Posts loading is correct.
- Albums loading is correct.
- Photos won't load sometimes.

As I understand it, this is due to the fact that all IDs fall into both requests. And when you receive posts, and when you receive albums, and when you receive photos. 
Why? Tell me, is this normal? If so, how do I get it loaded with the right data?

### What steps will reproduce the problem?

**api/models/NewsfeedSearch.php:**
```php
    public function search($params)
    {
        $query = Newsfeed::find()->active();

        $query->distinct();

        $query->joinWith([
            'post' => function (ActiveQuery $query) {
                if ($query->primaryModel !== null) {
                    $query->onCondition([self::tableName() . '.model' => StringHelper::basename(Post::className())]);
                    $query->andOnCondition(['post.spam' => 0]);
                } else {
                    $query->onCondition(['post.spam' => 0]);
                }
            },
            'album.photos' => function (ActiveQuery $query) {
                if ($query->primaryModel !== null) {
                    $query->onCondition([self::tableName() . '.model' => StringHelper::basename(Album::className())]);
                    $query->andOnCondition("to_date(to_char(to_timestamp(" . Photo::tableName() . ".created_at), 'yyyy-MM-dd'), 'yyyy-MM-dd') = " . self::tableName() . ".date");
                    $query->andOnCondition(['photo.spam' => 0]);
                } else {
                    $query
                        ->onCondition(['photo.spam' => 0])
                        ->orderBy(Photo::tableName() . ".created_at DESC, " . Photo::tableName() . ".id DESC")
                        ->limit(5);
                }
            }
        ]);

        $query->andWhere(['or', ['is not', 'post.spam', null], ['is not', 'photo.spam', null]]);

        $dataProvider = new ActiveDataProvider([
            'query' => $query,
        ]);

        $this->load($params, '');

        if (!$this->validate()) {
            return $this;
        }

        // Many other conditions ...

        $query->orderBy(self::tableName() . '.published_at DESC, ' . self::tableName() . '.id DESC');

        return $dataProvider;
    }
```

**api/models/Newsfeed.php:**
```php
public function getPost()
{
    return $this->hasOne(Post::className(), ['id' => 'model_id'])->with(['comments', 'copyPost']);
}

public function getAlbum()
{
    return $this->hasOne(Album::className(), ['id' => 'model_id'])->with('photos');
}
```

**api/models/Album.php:**
```php
public function getPhotos()
{
    return $this->hasMany(Photo::className(), ['id' => 'photo_id'])
        ->viaTable(AlbumPhoto::tableName(), ['album_id' => 'id'])
        ->orderBy('created_at DESC');
}
```

### What is the expected result?
News, consisting of posts, albums (which in turn contain up to 5 photos).

### What do you get instead?
News, consisting of posts, albums. Albums sometimes come true, sometimes they do not contain photos.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.13
| PHP version      | 7.0.22-2
| Operating system | ubuntu16.04.1

**Comment by cebe** [comment]
could be related to #5341 maybe even duplicate.

**Comment by machour** [comment]
Should be fixed in 2.0.16 by #15308
